### PR TITLE
Makes the Radio Station's record player an admin-toggleable feature.

### PR DESCRIPTION
**Pull request by CaioFrancisco**
## About the PR 
The PR makes a new button under Server Toggles that allows for Primary Admins+ to toggle the Radio Station's record player and tape deck.

## Why's this needed? 
This allows for a more RP-friendly or at least a less aggressive way for admins to play their songs without blowing up, desintegrating the record player or whatever shenanigans involving disabling the record player, and allows it to be turned on again!

### CODE REVIEW

**Comment by CaioFrancisco** [comment]
Can I ignore the linters going off about how I used an if statement on a constant var?

File: code/modules/telescience/radiostation.dm [hunk 1]
```diff
@@ -2,6 +2,9 @@
 //###~GANNETS' RADIO STATION STUFF~###//
 //####################################//
 
+// When this comment was written, the Radio Station records were disabled. Remove this pre-processor line and change the description on line 242 when it's enabled again.
+#define MUSIC_OUTOFORDER 1
```
Comment (pgmzeta) [inline code/modules/telescience/radiostation.dm:6]: `1` ->`TRUE`

File: code/modules/telescience/radiostation.dm [hunk 2]
```diff
@@ -236,7 +239,7 @@
 // Record player
 /obj/submachine/record_player
 	name = "record player"
-	desc = "An old school vinyl record player sat on a set of drawers. Shame you don't have any records."
+	desc = "An old school vinyl record player sat on a set of drawers. Shame you don't have any records. There's a small \"out of order\" label on the casing."
```
Comment (pgmzeta) [inline code/modules/telescience/radiostation.dm:242]: why not dynamically append the description if the define is true instead of having to make two changes?

File: code/modules/telescience/radiostation.dm [hunk 3]
```diff
@@ -254,6 +257,9 @@
 		if(has_record)
 			boutput(user, "The record player already has a record inside!")
 		else if(!is_playing)
+			if(MUSIC_OUTOFORDER)
```
Comment (pgmzeta) [inline code/modules/telescience/radiostation.dm:260]: why not stop the record from going in at all?

**Review by pgmzeta — COMMENTED** [review]
Not an admin, but I think it would be useful to have the option to disable this in-game in e.g. when events with music are run, not just at compile time. Not sure how long this is expected to be disabled.

**Comment by CaioFrancisco** [comment]
Set to draft so I can properly do the changes pgmzeta requested, since they sound reasonable and can serve for a multi-use and long-term change.

**Comment by CaioFrancisco** [comment]
This commit doesn't work as much as I wanted it to. The whole thing with the Record Player works perfectly, but I can't for the life of me make the Admin Panel button of disabling/enabling record players work. If anyone can give me an idea, I would be grateful.

**Comment by NightmareChamillian** [comment]
It's worth mentioning the CL won't automatically append with how it is right now- you may want to copy paste a fresh one from the template

**Comment by CaioFrancisco** [comment]
> It's worth mentioning the CL won't automatically append with how it is right now- you may want to copy paste a fresh one from the template

I'm afraid I don't know what the CL you are mentioning is. Can you put it in layman's terms?

**Comment by NightmareChamillian** [comment]
oh! the changelog.

**Comment by CaioFrancisco** [comment]
> oh! the changelog. The way it's set up is that there's a bot that goes through pull requests and looks for a comment box to pull the changelog additions from, then adds it to the html. Without the formatting it won't get auto-added to the popup window.

thanks for pointing out, i was afraid it was something i had to change in the code, but i can do that from my phone.

**Comment by CaioFrancisco** [comment]
Since music is back again, I also changed my code to make each record player start pre-activated, so it can play music again.

**Comment by CaioFrancisco** [comment]
Whoops, accidentally closed it. It's back open.

**Comment by CaioFrancisco** [comment]
I'll close this down and wait for the feature-freeze to end. Perhaps I'll make it actually work with admin panels in the mean time.

**Comment by CaioFrancisco** [comment]
It's back, working like intended. Took a bit more than I expected, to be honest, but it has most of the comfort features I wanted. Now there are two things I would code in but I don't have enough courage to add without confirmation: making whether Radio Audio (records and tapes) are on or off, so we don't have to display the message for it being disabled or enabled twice or more, and whether admins should be messaged whenever it is turned on or off. I'll ask around in #imcoder, but any admin input in here is very much appreciated.

**Comment by CaioFrancisco** [comment]
christ i fucking hate git. i'm sorry for everyone who got requested a review for being code owners.

**Comment by frawhst** [comment]
lolrip

**Comment by CaioFrancisco** [comment]
Shouldn't be a crime against humanity itself now. Sorry for the inconvenience.

File: code/modules/telescience/radiostation.dm [hunk 4]
```diff
@@ -789,12 +806,27 @@ ABSTRACT_TYPE(/obj/item/record/random/notaquario)
...
-	var/has_tape = 0
-	var/is_playing = 0
+	var/has_tape = FALSE
+	var/is_playing = FALSE
+	var/can_play_tapes = TRUE
 	var/obj/item/radio_tape/tape_inside = null
 
+	New()
+		. = ..()
+		START_TRACKING
```
Comment (ZeWaka) [inline code/modules/telescience/radiostation.dm:816]:
```suggestion
		START_TRACKING

```

File: code/modules/telescience/radiostation.dm [hunk 5]
```diff
@@ -241,17 +241,34 @@
...
+		if(!src.can_play_music)
+			. += " There's an \"out of order\" label on it."
+
+	disposing()
+		STOP_TRACKING
+		. = ..()
+
+	get_desc()
+		if(!src.can_play_music)
+			. += " There's an \"out of order\" label on it."
```
Comment (ZeWaka) [inline code/modules/telescience/radiostation.dm:264]: ?
```suggestion
```

File: code/modules/admin/admin.dm [hunk 6]
```diff
@@ -4300,7 +4352,10 @@ var/global/noir = 0
 				<A href='?src=\ref[src];action=secretsadmin;type=manifest'>Crew Manifest</A> |
 				<A href='?src=\ref[src];action=secretsadmin;type=DNA'>Blood DNA</A> |
 				<A href='?src=\ref[src];action=secretsadmin;type=fingerprints'>Fingerprints</A><BR>
-
+				Radio Music | <A href='?src=\ref[src];action=radio_audio_allow'>ON</A> |
+				<A href='?src=\ref[src];action=radio_audio_disallow'>OFF</A><BR>
+				Player Radio Records/Tapes | <A href='?src=\ref[src];action=radio_audio_allow'>ON</A> |
+				<A href='?src=\ref[src];action=radio_audio_disallow'>OFF</A><BR>
```
Comment (ZeWaka) [inline code/modules/admin/admin.dm:4358]: why is there two buttons that do the same thing

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/admin/admin.dm [hunk 7]
```diff
@@ -414,6 +414,24 @@ var/global/noir = 0
...
+				tgui_alert(usr,"You need to be at least a Moderator to allow radio audio.")
+		if("radio_audio_disallow")
+			if(src.level >= LEVEL_MOD)
+				for(var/obj/submachine/record_player/O in by_type[/obj/submachine/record_player])
+					O.can_play_music = FALSE
+				for(var/obj/submachine/tape_deck/O in by_type[/obj/submachine/tape_deck])
+					O.can_play_tapes = FALSE
+				logTheThing("admin", usr, null, "disallowed for radio music/tapes to play.")
+			else
+				tgui_alert(usr,"You need to be at least a Moderator to disallow radio audio.")
```
Comment (Tarmunora) [inline code/modules/admin/admin.dm:434]: any reason for this to not just be a toggle, instead of two separate commands?

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/telescience/radiostation.dm (hunk 5 above)
Comment (CaioFrancisco) [reply code/modules/telescience/radiostation.dm:264]: Another left-over from when I tried to fix rebasing issues. I'll remove it.

File: code/modules/admin/admin.dm (hunk 7 above)
Comment (CaioFrancisco) [reply code/modules/admin/admin.dm:434]: I considered making it a toggle, but I then realized that it could enter in a really weird state if someone tries to varedit a Record Player/Tape Deck. I'll try to make it a toggle without depending on each individual players though.

**Comment by CaioFrancisco** [comment]
I'm just waiting for it to finish building so I can test if it is good to be committed. One last question: should all the admins get messaged when someone toggles the record players and tape decks?

File: code/modules/admin/admin.dm (hunk 7 above)
Comment (CaioFrancisco) [reply code/modules/admin/admin.dm]: I made it toggleable in the new commit, but I'll leave this open to get some opinion on how I did it and how I should be doing it.

**Review by Tarmunora — CHANGES_REQUESTED** [review]
ok I think i was maybe unclear, but I meant doing it in a similar fashion to the other server_toggles (see `code/modules/admin/toggles.dm`), and being under `	SET_ADMIN_CAT(ADMIN_CAT_SERVER_TOGGLES)` etc

**Comment by CaioFrancisco** [comment]
That completely passed by my head. Should be mostly coded and optimized though, I couldn't do it the entire day because intensive school times.

**Comment by CaioFrancisco** [comment]
That should be all. The whole code has been transferred to Server Toggles.

**Comment by virvatuli-gh** [comment]
Yes please, this is something I've wanted for awhile.
